News and compatibility annotations for Julia 1.1

fredrikekre opened the pull request:
Me and @mortenpi have gone through the ~570 PRs that have been merged since Julia 1.0 (https://docs.google.com/spreadsheets/d/1PThPlhFwlkcmmEpe8JcJ-XyTdU-I1lmEh7iCw8AvFSM/edit?usp=sharing20) in preparation for the 1.1 release. We have added news and compatibility annotations for changes worthy of that.

Since we now have one doc for all of Julia 1.X (#26825) we have thus added small `!!! compat` notes to the manual for e.g. new functions and new behaviors. For a taste of what these annotations look like see e.g. https://julialang.github.io/Pkg.jl/dev/registries/

Walking through the PRs like this is not very efficient (and pretrty boring), so for future features I ask people to add these annotations together with NEWS.

StefanKarpinski commented:
OMG, you two are saints!

mortenpi commented:
> so for future features I ask people to add these annotations together with NEWS.

Should we add a note to [CONTRIBUTING.md](https://github.com/JuliaLang/julia/blob/master/CONTRIBUTING.md#news-worthy-changes)? Something along the lines of:

````
#### Annotations for new features, deprecations and behaviour changes

API additions and deprecations, and minor behavior changes are allowed in minor version releases. For documented features that are part of the public API, a compatibility note should be added into the manual or the docstring. It should state the Julia minor version that made changed the behavior and have a brief message describing the change.

At the moment, this should always be done with the following `compat` admonition (so that it would be possible to find the compat annotations programmatically in the future):

```
!!! compat "Julia 1.X"
    This method was added in Julia 1.X.
```
````

JeffBezanson commented:
This is amazing. I was just starting to worry about how we were going to do this for this release and who was going to do it (me?) but now here it is :)

StefanKarpinski commented:
I cannot emphasize how wonderful it is that you two did this. Thank you sooo much! ❤️❤️❤️

ararslan commented on NEWS.md:
```diff
@@ -59,10 +61,41 @@ Standard library changes
+External dependencies
+---------------------
+
+  * 7zip (bundled with Julia on Windows) have been upgraded from version 16.04 to 18.05 ([#30035]).
```
"Have" should be "has" here and for OpenBLAS and LLVM.

fredrikekre replied:
:sweden:

mbauman commented on base/multidimensional.jl:
```diff
@@ -211,6 +211,28 @@ module IteratorsMD
+    `CartesianIndices` support broadcasting arithmetic (+ and -) with a `CartesianIndex`.
+
+    !!! compat "Julia 1.1"
+    Broadcasting of CartesianIndices requires at least Julia 1.1.
```
Does this need to be indented?

mbauman commented:
It's not a very strong preference — which is why I figured I'd raise the possibility on slack instead of here.  My thought is just to differentiate things a bit more from warnings, but the green does a pretty good job of that.  And we also use the encircled ! for `!!! note`s, too, so let's not worry about the color of this 🚲🏠 too much.

mortenpi committed:
Addition of NEWS and compat admonitions for important
changes between Julia 1.0 and 1.1, including:

- Custom .css-style for compat admonitions.

- Information about compat annotations to CONTRIBUTING.md.

- NEWS.md entry for PRs #30090, #30035, #30022, #29978,
  #29969, #29858, #29845, #29754, #29638, #29636, #29615,
  #29600, #29506, #29469, #29316, #29259, #29178, #29153,
  #29033, #28902, #28761, #28745, #28708, #28696, #29997,
  #28790, #29092, #29108, #29782

- Compat annotation for PRs #30090, #30013, #29978,
  #29890, #29858, #29827, #29754, #29679, #29636, #29623,
  #29600, #29440, #29316, #29259, #29178, #29157, #29153,
  #29033, #28902, #28878, #28761, #28708, #28156, #29733,
  #29670, #29997, #28790, #29092, #29108, #29782, #25278

- Documentation for broadcasting CartesianIndices (#30230).
- Documentation for Base.julia_cmd().
- Documentation for colon constructor of CartesianIndices (#29440).
- Documentation for ^(::Matrix, ::Number) and ^(::Number, ::Matrix).

- Run NEWS-update.jl.

Co-authored-by: Morten Piibeleht <morten.piibeleht@gmail.com>
Co-authored-by: Fredrik Ekre <ekrefredrik@gmail.com>
